### DISCUSSION TITLE
airport forecast

### DISCUSSION THREAD

**Discussion by JensK99** [team-sx/iwxxm-sx#17] [Ideas]
A couple month ago I tried to design our airport forecast as a xml-format, wich looks like this:
![image](https://github.com/team-sx/iwxxm-sx/assets/141999281/3f01988f-537a-45da-9261-7da0f7773431)
I looked at this time on Choys WAFS-example and tried to implement it in this way. But the WAFS is not the airport forecast and I had to make some assumptions. I also did not use up to know other schematas to have it in an easy way.
The format looks like that:
```xml
<airportWX>
<aerodrome>
   <name>M&uuml;nchen</name>
   <locationIndicatorICAO>EDDM</locationIndicatorICAO>
   <ElevatedPoint srsDimension="2" axisLabels="Lat Long" srsName="http://www.opengis.net/def/crs/EPSG/0/4326">
      <gml:pos>48.353378 11.78609</gml:pos>
      <radius uom="nm">5</radius>
      <elevation uom="ft">1487</elevation><verticalReference>MSL</verticalReference>
   </ElevatedPoint>
</aerodrome>
<boundingPeriod>
   <TimePeriod>
      <beginPosition>2023-09-25T06:00:00Z</beginPosition>
      <endPosition>2023-09-26T03:00:00Z</endPosition>
    </TimePeriod>
</boundingPeriod>
<boundingVolume>
   <upperElevation>5000</upperElevation>
   <upperVerticalReference>MSL</upperVerticalReference>
   <lowerElevation>GND</lowerElevation>
   <lowerElevationReference>OTHER</lowerElevationReference>
</boundingVolume>
<featureCollection>
   <feature>
      <phenomenon>wind</phenomenon>
      <phenomenonProperty>
         <propertyName>direction</propertyName>
         <shortName>WD</shortName>
         <TimeSeries> 
            <forecastedTime>
               <timePosition>2023-09-25T06:00:00Z</timePosition>
               <value uom="deg">080</value>
            </forecastedTime>
            <forecastedTime>
               <timePosition>2023-09-25T07:00:00Z</timePosition>
               <value uom="deg">080</value>
            </forecastedTime>
            ....
         </TimeSeries> 
         <ElevatedPoint srsDimension="2" axisLabels="Lat Long" srsName="http://www.opengis.net/def/crs/EPSG/0/4326"><gml:pos>48.353378 11.78609</gml:pos><elevation uom="m">10</elevation><verticalReference>AGL</verticalReference></ElevatedPoint>
      </phenomenonProperty>
      <phenomenonProperty>
         <propertyName>speed</propertyName>
         <shortName>WS</shortName>
         <TimeSeries> 
            <forecastedTime>
               <timePosition>2023-09-25T06:00:00Z</timePosition>
               <value uom="kn">4</value>
            </forecastedTime>
             ....
         </TimeSeries> 
         <ElevatedPoint srsDimension="2" axisLabels="Lat Long" srsName="http://www.opengis.net/def/crs/EPSG/0/4326"><gml:pos>48.353378 11.78609</gml:pos><elevation uom="m">10</elevation><verticalReference>AGL</verticalReference></ElevatedPoint>
      </phenomenonProperty>
      ....
</feature>
....
 <feature>
      <phenomenon>vicinity </phenomenon>
      <phenomenonGeometry>
         <gml:patches>
            <gml:PolygonPatch>
               <gml:exterior>
                  <gml:LinearRing>
                     <gml:posList srsDimension="2">49.03333 11.78333 49.03333 13.18333 48.35000 13.18333 48.35000 11.78333 49.03333 11.78333</gml:posList>
                  </gml:LinearRing>
               </gml:exterior>
            </gml:PolygonPatch>
         </gml:patches>
      </phenomenonGeometry>
      <phenomenonProperty>
         <propertyName>northeast sector weather</propertyName>
         <shortName>VCNE</shortName>
         <TimeSeries> 
            <forecastedTime>
               <timePosition>2023-09-25T06:00:00Z</timePosition>
               <value>NIL</value>
            </forecastedTime>
            ....
         </TimeSeries> 
      </phenomenonProperty>
      <phenomenonGeometry>
         <gml:patches>
            <gml:PolygonPatch>
               <gml:exterior>
                  <gml:LinearRing>
                     <gml:posList srsDimension="2">49.03333 10.38333 49.03333 11.78333 48.35000 11.48333 48.35000 10.48333 49.03333 10.03333</gml:posList>
                  </gml:LinearRing>
               </gml:exterior>
            </gml:PolygonPatch>
         </gml:patches>
      </phenomenonGeometry>
      <phenomenonProperty>
         <propertyName>northwest sector weather</propertyName>
         <shortName>VCNW</shortName>
         <TimeSeries> 
            <forecastedTime>
               <timePosition>2023-09-25T06:00:00Z</timePosition>
               <value>NIL</value>
            </forecastedTime>
            ....
         </TimeSeries> 
      </phenomenonProperty>
      ....
</feature>
....
</featureCollection>
</airportWX>

```
When I designed the details i had to solve some problems. Choys rough expression on ticket https://github.com/team-sx/iwxxm-sx/issues/16 doesnt handle all the details. 

Problems:
* what should be a feature and what should be a property? For example I defined that wind is a feature and wind speed will be a property.
* radius around the airport in GML not possible?
* I have designed it that the Feature can have a geometry but not the property can have a geometry. So I have a problem in the vicinity feature when I define the different sectors as properties as they have a different geometry
* first I designed it like that, that one property can have one value, but in some cases there are more than one value. For example a min and a max value, so I added a tag `<valueMin>` and `<valueMax>` but there might be some other cases where it will be some totally different values

**discussion content not yet finished**

**Comment by blchoy** [comment]
My suggested development stages will be:

1. Create WxObjects of the phenomena you would like to describe.  [This discussion](https://github.com/wmo-im/iwxxm/wiki/VONA-And-QVACI-Design-Review-Teleconference-2023-Nov-2) may give you some hints on how to do this in describing a volcano, a volcanic ash cloud (under the VONA context) and another volcanic ash cloud (under QVA context).
2. Now you may want to say that why don't we create a volcanic ash cloud that can be used for both VONA and QVA.  In fact the VONA TAC template limits description of the properties in association with a volcanic ash cloud.  The reverse is true as we are using the descriptions of a volcano in VONA for QVA.
3. Depending on the story you would like to tell put all the necessary WxObjects into an iwxxm:MeteorologicalFeatureCollection (in fact an extension to the type) together with additional elements specific to the collection but not the WxObjects.  Again the above discussion may give you some clue on how to do this.

In creating a WxObject you will need to specify its geometry and its properties.  You can either used predefined geometries like ElevatedVolume etc. or you can define your own.  For properties we have yet to define generic ones to be shared so far, but it would not be difficult to do so.

So a wind WxObject may look like:

```
<iwxxm:MeteorologicalFeature .../>
    <gml:identifier .../>
    <iwxxm:issueTime .../>
    <iwxxm:originatingCentre .../>
    <iwxxm:phenomenonCategory>Observations</iwxxm:phenomenonCategory>
    <iwxxm:phenomenonTime .../>
    <iwxxm:phenomenon xlink:href="https://codes.wmo.int/49-2/MeteorologicalFeature/WIND"/>
    <iwxxm:phenomenonGeometry>
        <!-- Description of a disc of 8km radius centred at the ARP of the aerodrome -->
    </iwxxm:phenomenonGeometry>
    <iwxxm:phenomenonProperty>
        <!-- Description of the wind measurement -->
    </iwxxm:phenomenonProperty>
</iwxxm:MeteorologicalFeature>
```

So if we have defined descriptions of wind  measurement (we did in the design of IWXXM METAR/SPECI), all we need to do is to see if we could generalize and enrich/simplify the content it carries.

**Reply by JensK99** [reply]
If I follow "this discussion" and examples then I see that I have to use the codelist https://codes.wmo.int/49-2/MeteorologicalFeature. But the things I want to express in the xml are not possible with this codelist.  This leads to the problem which I tried to describe "What should be feature ad what should be a property?".

**Reply by blchoy** [reply]
Of course we can add new stuff to this codelist, and that is why we come together as we need to determine the taxonomy of this codelist.
